build.rs

```diff
@@ -113,10 +113,16 @@ fn compile_freebsd() {
 }
 
 fn compile_windows() {
-    cc::Build::new()
+    let mut cc = cc::Build::new();
+    cc
         .file("etc/hidapi/windows/hid.c")
-        .include("etc/hidapi/hidapi")
-        .compile("libhidapi.a");
+        .include("etc/hidapi/hidapi");
+
+    if cfg!(target_feature = "crt-static") {
```

Comment:
I think it is not valid to use *cfg!* macro or the *cfg* attribure for this, it may break cross compilation. build.rs compilation is detached from the main library build.

You can access the *target_feature* of the library from the build.rs with the *CARGO_CFG_TARGET_FEATURE* environment variable.

See e.g. https://github.com/google/shaderc-rs/blob/647e07510c160f6d15e63edd6c4f089e353cef76/shaderc-sys/build/build.rs#L84 for an example.

Reply:
> cross compilation

Hmm.. I saw both ways, actually. You're using the env variable, I know. I thought, they are interchangeable. Is there any link to dig about cross-compilation in cargo?

Reply:
Check db2065a

Reply:
I am sure, it is mentioned somewhere in the reference.
If you think about it, the build.rs can not really knowm which cfgs are enabled with cfg!, because that is resolved at compile time of the build.rs. The needed information is however passed into the resulting build.rs executable at runtime.